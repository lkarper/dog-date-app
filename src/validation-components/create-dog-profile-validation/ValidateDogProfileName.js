import React, { useEffect } from 'react';

const ValidateDogProfileName = (props) => {

    const { name, nameError, setNameError } = props;

    useEffect(() => {
        if (name.trim().length === 0) {
            setNameError(`A name is required. (Max 72 characters.)`);
        } else {
            setNameError('');
        }
    }, [name, setNameError]);

    if (nameError) {
        return (
            <p 
                className='ValidateDogProfile__validator error'
                id='name-validator'
            >
                {nameError}
            </p>
        );
    }

    return (
        <p 
            className='ValidateDogProfile__validator valid'
            id='name-validator'
        >
            Name accepted. (Max 72 characters.)
        </p>
    );
    
}

export default ValidateDogProfileName;
