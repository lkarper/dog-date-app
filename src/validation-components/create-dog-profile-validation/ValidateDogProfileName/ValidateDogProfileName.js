import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const ValidateDogProfileName = (props) => {

    const { 
        name, 
        nameError, 
        setNameError 
    } = props;

    useEffect(() => {
        if (name.trim().length === 0) {
            setNameError(`A name is required. (Max 72 characters.)`);
        } else {
            setNameError('');
        }
    }, [name, setNameError]);

    if (nameError) {
        return (
            <p 
                className='ValidateDogProfile__validator error'
                id='name-validator'
            >
                {nameError}
            </p>
        );
    }

    return (
        <p 
            className='ValidateDogProfile__validator valid'
            id='name-validator'
        >
            Name accepted. (Max 72 characters.)
        </p>
    );
    
}

ValidateDogProfileName.defaultProps = {
    name: '',
    nameError: 'A name is required. (Max 72 characters.)',
    setNameError: () => {},
};

ValidateDogProfileName.propTypes = {
    name: PropTypes.string.isRequired,
    nameError: PropTypes.string.isRequired,
    setNameError: PropTypes.func.isRequired,
};

export default ValidateDogProfileName;
