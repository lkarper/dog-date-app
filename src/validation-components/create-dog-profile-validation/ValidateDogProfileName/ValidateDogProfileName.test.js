import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateDogProfileName from './ValidateDogProfileName';

describe('ValidateDogProfileName component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateDogProfileName />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateDogProfileName />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
