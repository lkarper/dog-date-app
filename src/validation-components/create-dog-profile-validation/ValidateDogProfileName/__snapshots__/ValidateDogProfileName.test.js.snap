// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateDogProfileName component renders the UI as expected 1`] = `
<ValidateDogProfileName
  name=""
  nameError="A name is required. (Max 72 characters.)"
  setNameError={[Function]}
>
  <p
    className="ValidateDogProfile__validator error"
    id="name-validator"
  >
    A name is required. (Max 72 characters.)
  </p>
</ValidateDogProfileName>
`;
